src/xml_parser.cc

```diff
@@ -63,14 +63,21 @@ xmlpp::Document* XMLParser::Document() {
   // This adds the capability to have nice include semantics
   bool generate_xinclude_nodes = true;
   bool fixup_base_uris = false;
-  #if LIBXMLXX_MAJOR_VERSION == 2 
-  doc->process_xinclude(generate_xinclude_nodes);
+  #if LIBXMLXX_MAJOR_VERSION < 2
```

Comment:
When does this happen?  Or is it just for completeness?

Reply:
On 24.04 `test_include_recipe.py` would fail.  You might remember that when libxml++ processes XIncludes it generates a `base` attribute (I saw you made some commits to this last summer/fall) which renders the xml invalid according to our RelaxNG schema.  There is logic in place to parse through and remove this attribute, but after v2.42.0 there is an argument to `process_xinclude()` to avoid generating the `base` attribute altogether.  See the [changelog with that release](https://github.com/libxmlplusplus/libxmlplusplus/releases/tag/2.42.0).

On ubuntu 22.04 the version of libxml++ is 2.40, and on our conda builds we use libxml++ 4.0

Reply:
Sorry - my question wasn't clear - when do we ever get libxml++ version < 2?

Reply:
Oh sorry, tracking now. It'll never be <2. The version states we need to check for are <2.42 and >2.42, if I'm not mistaken. I will clean up that logic

Reply:
Should be good now.